sdk/servicebus/Azure.Messaging.ServiceBus/tests/Samples/Sample15_MockingClientTypes.cs

```diff
@@ -0,0 +1,665 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Messaging.ServiceBus.Administration;
+using Moq;
+using Moq.Protected;
+using NUnit.Framework;
+using NUnit.Framework.Constraints;
+
+namespace Azure.Messaging.ServiceBus.Tests.Samples
+{
+    public class Sample15_MockingClientTypes : ServiceBusLiveTestBase
+    {
+        [Test]
+        public async Task MockSendToQueue()
+        {
+            #region Snippet:ServiceBus_MockingSendToQueue
+
+            Mock<ServiceBusClient> mockClient = new();
+            Mock<ServiceBusSender> mockSender = new();
+
+            mockClient
+                .Setup(client =>client.CreateSender(It.IsAny<string>()))
+                .Returns(mockSender.Object);
+
+            mockSender
+                .Setup(sender => sender.SendMessageAsync(
+                    It.IsAny<ServiceBusMessage>(),
+                    It.IsAny<CancellationToken>()))
+                .Returns(Task.CompletedTask);
+
+            ServiceBusClient client = mockClient.Object;
+
+            // The rest of this snippet illustrates how to send a service bus message using the mocked
+            // service bus client above, this would be where application methods sending a message would be
+            // called.
+
+            var mockQueueName = "MockQueueName";
+            ServiceBusSender sender = client.CreateSender(mockQueueName);
+            ServiceBusMessage message = new ServiceBusMessage("Hello world!");
+
+            await sender.SendMessageAsync(message);
+
+            #endregion
+        }
+
+        [Test]
+        public async Task MockSendBatch()
+        {
+            #region Snippet:ServiceBus_MockingSendBatch
+
+            Mock<ServiceBusClient> mockClient = new();
+            Mock<ServiceBusSender> mockSender = new();
+
+            mockClient
+                .Setup(client => client.CreateSender(It.IsAny<string>()))
+                .Returns(mockSender.Object);
+
+            List<ServiceBusMessage> backingList = new List<ServiceBusMessage>();
+
+            int batchCountThreshold = 5;
+
+            ServiceBusMessageBatch mockBatch = ServiceBusModelFactory.ServiceBusMessageBatch(
+                batchSizeBytes: 500,
+                batchMessageStore: backingList,
+                batchOptions: new CreateMessageBatchOptions(),
+                tryAddCallback: message =>
```

Comment:
nit:
```suggestion
                tryAddCallback: _=> backingList.Count < batchCountThreshold
```

Reply:
Woops looks like I didn't do the suggestion right because it still includes the subsequent lines that should be deleted.